packages/dd-trace/test/llmobs/util.js

```diff
@@ -10,144 +14,229 @@ const MOCK_NUMBER = Symbol('number')
-    } else if (expected[key] === MOCK_NUMBER) {
-      new chai.Assertion(typeof actual[key], `key ${key}`).to.equal('number')
-    } else if (expected[key] === MOCK_OBJECT) {
-      new chai.Assertion(typeof actual[key], `key ${key}`).to.equal('object')
-    } else if (expected[key] === MOCK_ANY) {
-      new chai.Assertion(actual[key], `key ${key}`).to.exist
-    } else if (Array.isArray(expected[key])) {
-      assert.ok(Array.isArray(actual[key]), `key "${key}" is not an array`)
-      const sortedExpected = [...expected[key].sort()]
-      const sortedActual = [...actual[key].sort()]
-      new chai.Assertion(sortedActual, `key: ${key}`).to.deepEqualWithMockValues(sortedExpected)
-    } else if (typeof expected[key] === 'object') {
-      new chai.Assertion(actual[key], `key: ${key}`).to.deepEqualWithMockValues(expected[key])
-    } else {
-      new chai.Assertion(actual[key], `key: ${key}`).to.equal(expected[key])
+/**
+ * @typedef {{
+ *   spanKind: 'llm' | 'embedding' | 'agent' | 'workflow' | 'task' | 'tool' | 'retrieval',
+ *   name: string,
+ *   inputMessages: { [key: string]: any },
+ *   outputMessages: { [key: string]: any },
+ *   inputDocuments: { [key: string]: any },
+ *   outputDocuments: { [key: string]: any },
+ *   inputValue: { [key: string]: any },
+ *   outputValue: { [key: string]: any },
+ *   metrics: { [key: string]: number },
+ *   metadata: { [key: string]: any },
+ *   modelName?: string,
+ *   modelProvider?: string,
+ *   parentId?: string,
+ *   error?: { message: string, type: string, stack: string },
+ *   span: unknown,
+ *   sessionId?: string,
+ *   tags: { [key: string]: any },
+ *   traceId?: string,
+ * }} ExpectedLLMObsSpanEvent
+ */
+
+/**
+ *
+ * @param {ExpectedLLMObsSpanEvent} expected
+ * @param {*} actual
+ * @param {string} key name to associate with the assertion
+ */
+function assertWithMockValues (actual, expected, key) {
+  const actualWithName = key ? `Actual (${key})` : 'Actual'
+
+  if (expected === MOCK_STRING) {
+    assert.equal(typeof actual, 'string', `${actualWithName} (${util.inspect(actual)}) is not a string`)
+  } else if (expected === MOCK_NUMBER) {
+    assert.equal(typeof actual, 'number', `${actualWithName} (${util.inspect(actual)}) is not a number`)
+  } else if (expected === MOCK_OBJECT) {
+    assert.equal(typeof actual, 'object', `${actualWithName} (${util.inspect(actual)}) is not an object`)
+  } else if (expected === MOCK_ANY) {
+    assert.ok(actual != null, `${actualWithName} does not exist`)
+  } else if (Array.isArray(expected)) {
+    assert.ok(Array.isArray(actual), `${actualWithName} (${util.inspect(actual)}) is not an array`)
+    assert.equal(
+      actual.length,
+      expected.length,
+      `${actualWithName} has different length than expected (${actual.length} !== ${expected.length})`
+    )
+
+    const sortedExpected = [...expected.sort()]
+    const sortedActual = [...actual.sort()]
```

Comment:
This still feels a bit off to me. I would expect numbers to have a semantical meaning in its order. Non-primitive entries will not be sorted anyway.